src/Arcus.Messaging.Pumps.ServiceBus/AzureServiceBusMessagePump.cs

```diff
@@ -0,0 +1,296 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Arcus.Messaging.Abstractions;
+using Arcus.Messaging.Pumps.Abstractions;
+using GuardNet;
+using Microsoft.Azure.ServiceBus;
+using Microsoft.Azure.ServiceBus.Core;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+
+namespace Arcus.Messaging.Pumps.ServiceBus
+{
+    public abstract class AzureServiceBusMessagePump<TMessage> : MessagePump<TMessage, AzureServiceBusMessageContext>
+    {
+        private bool _isHostShuttingDown;
+        private MessageReceiver _messageReceiver;
+        private readonly MessageHandlerOptions _messageHandlerOptions;
+        private readonly AzureServiceBusMessagePumpSettings _messagePumpSettings;
+
+        /// <summary>
+        ///     Constructor
+        /// </summary>
+        /// <param name="configuration">Configuration of the application</param>
+        /// <param name="serviceProvider">Collection of services that are configured</param>
+        /// <param name="logger">Logger to write telemetry to</param>
+        protected AzureServiceBusMessagePump(IConfiguration configuration, IServiceProvider serviceProvider,
+            ILogger logger)
+            : base(configuration, serviceProvider, logger)
+        {
+            _messagePumpSettings = serviceProvider.GetRequiredService<AzureServiceBusMessagePumpSettings>();
+            _messageHandlerOptions = DetermineMessageHandlerOptions(_messagePumpSettings);
+        }
+
+        /// <summary>
+        ///     Path of the entity to process
+        /// </summary>
+        public string EntityPath { get; private set; }
+
+        /// <summary>
+        ///     Service Bus namespace that contains the entity
+        /// </summary>
+        public string Namespace { get; private set; }
+
+        /// <inheritdoc />
+        protected override async Task ExecuteAsync(CancellationToken stoppingToken)
+        {
+            try
+            {
+                _messageReceiver = await CreateMessageReceiverAsync(_messagePumpSettings);
+
+                Logger.LogInformation("Starting message pump on entity path '{EntityPath}' in namespace '{Namespace}'",
```

Comment:
Yeah, something I had to think of. The entity parh/namespace is something in the connection string that is secretly passed allong; but here we publicly write it to the log. Is that OK?

Reply:
That's fine, entity in connection string is because the authentication is only scoped to that entity itself and the key won't work with namespace/other entities

https://docs.microsoft.com/en-us/azure/service-bus-messaging/service-bus-authentication-and-authorization#shared-access-signature